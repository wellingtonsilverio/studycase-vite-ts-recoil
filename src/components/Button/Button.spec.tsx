import { render, screen } from "@testing-library/react";
import Button from "./";

describe("Button Component", () => {
    it("should render component", () => {
        const { container } = render(<Button />);

        expect(container).toMatchSnapshot();
    });
    
    it("should render text in button", () => {
        render(<Button>Hi</Button>);

        expect(screen.findAllByText("Hi")).toBeInTheDocument();
    });
    
    it("should onClick call function", () => {
        const ff = jest.fn();

        render(<Button onClick={ff}>Hi</Button>);

        expect(ff).toHaveBeenCalled();
    });
});