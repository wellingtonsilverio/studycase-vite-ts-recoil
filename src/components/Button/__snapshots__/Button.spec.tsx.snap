// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render component 1`] = `
<div>
  <button />
</div>
`;
